### ISSUE TITLE
Sentry-cli versions 2.x are missing Content-Type header

### ISSUE THREAD

**Issue by iivoraitahila** [getsentry/sentry-cli#1370]
### Environment

Sentry 22.9.0 self hosted
Sentry-cli 2.8.0
Sentry-cli 1.74.x

### Steps to Reproduce

1. curl -sL https://sentry.io/get-cli/ | SENTRY_CLI_VERSION=2.8.0 bash
2. Set SENTRY_DSN environment variable
3. sentry-cli send-event -m "testing"
4. returns "Event dispatched: bb4c19fc-d1fc-4cc4-a190-c4eb3846fc71"
5. event not visible in Sentry, not a trace

1. curl -sL https://sentry.io/get-cli/ | SENTRY_CLI_VERSION=1.7.4 bash
2. Set SENTRY_DSN environment variable
3. sentry-cli send-event -m "testing"
4. returns "Event dispatched: b00bfc94-f764-4f76-b17b-b4aadc2f63b0"
5. event is visible in Sentry

### Expected Result

Event to be visible in Sentry or an error message from Sentry-cli

### Actual Result

I checked with Wireshark that the 2.x versions get a 403 error response from Sentry and do not report this to the user. The difference between 2.x and 1.x is that 1.x includes a "Content-Type: application/json" header in the request. I was able to create a workaround by having a HTTP proxy in front of Sentry that adds the "Content-Type: application/json" header to the request.

**Comment by kamilogorek** [comment]
This is correct behavior, as v2 is not sending a JSON, but rather an "envelope" - https://develop.sentry.dev/sdk/envelopes/

**Comment by iivoraitahila** [comment]
:disappointed: What kind of configuration should I add? I'm just running the install.sh on a fresh Ubuntu virtual machine. Did you use a self-hosted instance on your test?
